src/logging/container.ts

```diff
@@ -0,0 +1,98 @@
+import * as es from 'event-stream';
+import { EventEmitter } from 'events';
+import * as _ from 'lodash';
+import * as Stream from 'stream';
+import StrictEventEmitter from 'strict-event-emitter-types';
+
+import Docker from '../lib/docker-utils';
+
+export interface ContainerLog {
+	message: string;
+	timestamp: number;
+	isStdout: boolean;
+}
+
+interface LogsEvents {
+	log: ContainerLog;
+	closed: void;
+	error: Error;
+}
+
+type LogsEventEmitter = StrictEventEmitter<EventEmitter, LogsEvents>;
+
+export class ContainerLogs extends (EventEmitter as {
+	new (): LogsEventEmitter;
+}) {
+	public constructor(public containerId: string, private docker: Docker) {
+		super();
+	}
+
+	public async attach(lastSentTimestamp: number) {
+		const logOpts = {
+			follow: true,
+			timestamps: true,
+			since: Math.floor(lastSentTimestamp / 1000),
+		};
+		const stdoutLogOpts = _.merge({ stdout: true, stderr: false }, logOpts);
+		const stderrLogOpts = _.merge({ stderr: true, stdout: false }, logOpts);
```

Comment:
```typescript
const stdoutLogOpts = { stdout: true, stderr: false, ...logOpts};
const stderrLogOpts = { stderr: true, stdout: false, ...logOpts};
```

Reply:
Same here, I expect the assign behaviour is what is wanted but it's worth checking as this isn't a 1-1 with `_.merge`

Reply:
I always forget about that syntax, but yeah this is what I want.